### PR TITLE
shady-css-scoped element package

### LINKED ISSUE #55: [ShadyCSS] Add standalone <custom-style> like element for scoping main document styles
We've had a request from a few people to be able to add scope-respecting main-document styles when not using Polymer, like with LitElement.

### PR DESCRIPTION
**Pull request by jrobinson01**
Adds a new package to expose a <custom-styles> element that wraps the custom-styles-interface polyfill.

### Reference Issue
fixes #55

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
I know the element is historically called custom-style, but is it worth considering giving this a more descriptive name now?

**Comment by justinfagnani** [comment]
I think it definitely should have a different name. It's really just a `<style>` element that respects emulated scoped. Since it's tied to ShadyCSS, it could have `shady` in the name. How about:

```html
<shady-css-scoped>
  <style>/* ... */</style>
</shady-css-scoped>
`?

**Comment by justinfagnani** [comment]
cc @azakus @sorvell

**Comment by jrobinson01** [comment]
I'm fine with a different name. I honestly don't even know what all it does aside from solve my problem of css theme variables not working in IE11 :). Given that, I vote for a name that helps describe what it does if possible.

**Comment by jrobinson01** [comment]
are we good with "shady-css-scoped" then?

**Comment by LarsDenBakker** [comment]
Yes

**Comment by jrobinson01** [comment]
Can we get this merged?

File: packages/shady-css-scoped-element/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# License
+
+Everything in this repo is BSD style license unless otherwise specified.
+
+Copyright (c) 2015 The Polymer Authors. All rights reserved.
```
Comment (azakus) [inline packages/shady-css-scoped-element/LICENSE.md]: Year should be 2019

File: packages/shady-css-scoped-element/src/shady-css-scoped-element.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/**
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
```
Comment (azakus) [inline packages/shady-css-scoped-element/src/shady-css-scoped-element.js]: License year should be 2019

File: packages/shady-css-scoped-element/src/shady-css-scoped-element.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+/*
+Wrapper over <style> elements to co-operate with ShadyCSS
+
+Example:
+<custom-style>
```
Comment (azakus) [inline packages/shady-css-scoped-element/src/shady-css-scoped-element.js]: Should be `<shady-css-scoped>`

File: packages/shady-css-scoped-element/tests/index.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<!--
+    @license
+    Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
```
Comment (azakus) [inline packages/shady-css-scoped-element/tests/index.html]: Year should be 2019

File: packages/shady-css-scoped-element/tests/shady-css-scoped.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        background-color: var(--foo);
+      }
+    </style>
+    <div id="test-div"></div>
+  </template>
+  <my-element></my-element>
+
+  <script>
+    suite('shady-css-scoped-element', function() {
+      setup(function() {
```
Comment (azakus) [inline packages/shady-css-scoped-element/tests/shady-css-scoped.html]: Use `suiteSetup`. `setup` runs before every test, so future tests will fail trying to define this element multiple times.

File: packages/shady-css-scoped-element/tests/shady-css-scoped.html [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+          connectedCallback() {
+            this.attachShadow({mode: 'open'});
+            this.shadowRoot.appendChild(document.querySelector('template').content.cloneNode(true));
+          }
+        }
+        customElements.define('my-element', MyElement);
+      });
+      test('css vars are applied to custom elements', function() {
+        var myElement = document.querySelector('my-element');
```
Comment (azakus) [inline packages/shady-css-scoped-element/tests/shady-css-scoped.html]: Use `const`

File: packages/shady-css-scoped-element/tests/shady-css-scoped.html [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+    The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+    The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+    Code distributed by Google as part of the polymer project is also
+    subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<html>
+<head>
+  <title>HTML Imports Dynamic Elements</title>
+  <script src="../node_modules/@webcomponents/webcomponentsjs/custom-elements-es5-adapter.js"></script>
```
Comment (azakus) [inline packages/shady-css-scoped-element/tests/shady-css-scoped.html]: Include the template polyfill for IE11 testing

**Comment by jrobinson01** [comment]
Can we get this published to npm?

**Comment by kevinpschaaf** [comment]
This PR broke tests on master; the tests here fail on polyfilled browsers because of the `customElements` dependency (external PR's don't test against the full suite of supported browsers on Sauce Labs, so this slipped through). You can see the failures in an (unrelated) CI log here: https://travis-ci.org/webcomponents/polyfills/builds/599913173#L935

Can someone (@jrobinson01 ?) work out adding the custom-elements polyfill to the tests and make a new PR?

**Comment by kevinpschaaf** [comment]
Actually, since this is blocking a release for other reasons, I'll look into fixing it.

**Comment by jrobinson01** [comment]
I’ll try to look tonight as well. I assume there are examples in other packages of what needs to be done (just mock customElements?)?

**Comment by kevinpschaaf** [comment]
@jrobinson01 Sorry, just saw your note. I put a PR up here: https://github.com/webcomponents/polyfills/pull/221

**Pull request merged**